### DISCUSSION TITLE
É possível executar um arquivo gulp de um diretório externo?

### DISCUSSION THREAD

**Discussion by luizangelomedeiros** [frontendbr/forum#880] [Perguntas e Respostas]
Olá galera, estou com dificuldade com o seguinte cenário.
Possuo dois repositórios locais de um projeto que faço parte, no "repositório 1" foi abstraído um conjunto de scripts usando gulp que tem como objetivo baixar um conteúdo via rest. No "repositório 2" tenho permissão de modificação somente de uma pasta.

Necessito executar o comando do gulp do "repositório 1" na pasta do "repositório 2" e que seu conteúdo fique nesta pasta . Exemplo:

C:\ Documentos\repositorio2 
$_ gulp ..\repositorio1 taskX --exemplo

Não sei se fui claro, mas seria mais menos isso.

PS: O erro que dá é que ele tenta executar o gulp na pasta que está, no caso a "repositorio2" e como não tem arquivo gulpfile , ele da a msg de instalar o gulp.

PS2: Fiz os comandos aqui mesmo "no braço" e são somente exemplos, ignorem se a sintaxe tiver errada hahaha

**Comment by felipefialho** [comment]
Já tentou copiar o `gulpfile` pra dentro do `repositorio2` usando o `postinstall` do npm scripts?

**Comment by herberthenrique** [comment]
Já tentou utilizar o comando cli passando o argumento `gulpfile` ?
`gulp --gulpfile <gulpfile path> <other_options>`

Acho que no caso do seu exemplo, ficaria algo mais ou menos assim:
` gulp --gulpfile ../repositorio1/gulpfile.js taskX --exemplo`

Não cheguei a testar, qualquer coisa da um toque :)

**Comment by luizangelomedeiros** [comment]
@herberthenrique valeeu pela dica.
Ele até "funciona", porém o que acontece é que ele muda o cwd, dai ele executa dentro do "repositório1".  Assim o conteúdo fica no "repositório1" e não no "repositório2".

```
C:\ Documents\repositorio2
gulp -f ..\repositorio1\gulpfile.js taskX --exemplo
[15:37:00] Working directory changed to ~\Documents\repositorio1
[15:37:01] Using gulpfile ~\Documents\repositorio1
[15:37:01] Starting 'taskX'...
```
Acredito que não tem como eu setar a pasta de destino do gulp né?
Exemplo:
`gulp -f ..\repositorio1\gulpfile.js --dest ..\repositorio2 taskX --exemplo`

Acho que não consigo mudar isso direto nas tasks, pq esse gulpfile vai ser usado por varias pessoas e fica complicado saber aonde colocaram o "repositorio2". 


@LFeh Booa ideia
É uma alternativa, porém eu teria a necessidade de algo bem parecido com a resposta do @herberthenrique , pois possuo varias dependências externas e arquivos de configuração dentro da pasta `repositorio1`.  Dai teria que alterar o gitignore para estes arquivos não serem comitados. Acredito que se não achar nenhuma solução que envolva "somente um comando" , o mais certo seria aplicar a que vc falou.

Valeeu pelo help galera :smile:

**Comment by felipefialho** [comment]
Maneiro, avisa quando resolver e a solução ;)

**Comment by klarkc** [comment]
n sei se se entendi bem o problema, mas talvez o [yargs](https://www.npmjs.com/package/yargs) lhe ajude, dai vc modifica o código da pasta de destino passando via parâmetro a pasta de destino.
 
Outra opção: https://github.com/gulpjs/gulp/blob/master/docs/recipes/specifying-a-cwd.md
